File: scripts/scripts/testing/automations/batch/estonia.R
url <- "https://raw.githubusercontent.com/okestonia/koroonakaart/master/koroonakaart/src/data.json"
df <- fromJSON(file = url)
df <- data.table(
    Date = df$dates2,
    `Cumulative total` = df$dataCumulativeTestsChart$testsAdminstered
)

df[, Country := "Estonia"]
df[, Units := "tests performed"]
df[, `Source URL` := "https://www.koroonakaart.ee/en"]
df[, `Source label` := "Central Health Information System and Patient Portal"]
df[, Notes := NA_character_]
df[, `Testing type` := "PCR only"]

fwrite(df, "automated_sheets/Estonia.csv")

File: scripts/scripts/testing/automations/batch/estonia.py
import pandas as pd


def main():

    url = "https://opendata.digilugu.ee/opendata_covid19_test_county_all.csv"

    df = pd.read_csv(url, usecols=["StatisticsDate", "TotalTests"])

    df = df.groupby("StatisticsDate", as_index=False).sum().sort_values("StatisticsDate")

    df = df.rename(columns={"StatisticsDate": "Date", "TotalTests": "Cumulative total"})

    df.loc[:, "Country"] = "Estonia"
    df.loc[:, "Units"] = "tests performed"
    df.loc[:, "Source URL"] = "https://www.terviseamet.ee/et/koroonaviirus/avaandmed"
    df.loc[:, "Source label"] = "Estonian Health Board"
    df.loc[:, "Notes"] = pd.NA

    df.to_csv("automated_sheets/Estonia.csv", index=False)


if __name__ == '__main__':
    main()
